src/main/scala/org/broadinstitute/dsde/workbench/sam/google/DisableUsersMonitor.scala

```diff
@@ -0,0 +1,189 @@
+import org.broadinstitute.dsde.workbench.google.GooglePubSubDAO
+import org.broadinstitute.dsde.workbench.google.GooglePubSubDAO.PubSubMessage
+import org.broadinstitute.dsde.workbench.model._
+import org.broadinstitute.dsde.workbench.sam._
+import org.broadinstitute.dsde.workbench.sam.util.SamRequestContext
+import org.broadinstitute.dsde.workbench.util.FutureSupport
+import spray.json._
+import io.opencensus.scala.Tracing
+import org.broadinstitute.dsde.workbench.model.WorkbenchIdentityJsonSupport.WorkbenchUserIdFormat
+import org.broadinstitute.dsde.workbench.sam.service.UserService
+
+import scala.concurrent.Future
+import scala.concurrent.duration.{FiniteDuration, _}
+
+/**
+  * Created by srubenst on 02/04/21.
+  */
+object DisableUsersMonitorSupervisor {
+  sealed trait DisableUsersMonitorSupervisorMessage
+  case object Init extends DisableUsersMonitorSupervisorMessage
+  case object Start extends DisableUsersMonitorSupervisorMessage
+
+  def props(
+             pollInterval: FiniteDuration,
+             pollIntervalJitter: FiniteDuration,
+             pubSubDao: GooglePubSubDAO,
+             pubSubTopicName: String,
+             pubSubSubscriptionName: String,
+             workerCount: Int,
+             userService: UserService
+           ): Props =
+    Props(new DisableUsersMonitorSupervisor(pollInterval, pollIntervalJitter, pubSubDao, pubSubTopicName, pubSubSubscriptionName, workerCount, userService))
+}
+
+class DisableUsersMonitorSupervisor(
+  val pollInterval: FiniteDuration,
+  pollIntervalJitter: FiniteDuration,
+  pubSubDao: GooglePubSubDAO,
+  pubSubTopicName: String,
+  pubSubSubscriptionName: String,
+  workerCount: Int,
+  userService: UserService
+) extends Actor with LazyLogging {
+  import DisableUsersMonitorSupervisor._
+  import context._
+
+  self ! Init
+
+  override def receive = {
+    case Init => init pipeTo self
+    case Start => for (i <- 1 to workerCount) startOne()
+    case Status.Failure(t) => logger.error("error initializing disable users monitor", t)
+  }
+
+  def init =
+    for {
+      _ <- pubSubDao.createSubscription(pubSubTopicName, pubSubSubscriptionName)
+    } yield Start
+
+  def startOne(): Unit = {
+    logger.info("starting DisableUsersMonitorActor")
+    actorOf(DisableUsersMonitor.props(pollInterval, pollIntervalJitter, pubSubDao, pubSubSubscriptionName, userService))
+  }
+
+  override val supervisorStrategy =
+    OneForOneStrategy() {
+      case e =>
+        logger.error("unexpected error in disable users monitor", e)
+        // start one to replace the error, stop the errored child so that we also drop its mailbox (i.e. restart not good enough)
+        startOne()
+        Stop
+    }
+
+}
+
+object DisableUsersMonitor {
+  case object StartMonitorPass
+
+  sealed abstract class DisableUserResult(ackId: String)
+  final case class ReportMessage(value: Map[WorkbenchEmail, Seq[SyncReportItem]], ackId: String) extends DisableUserResult(ackId = ackId)
```

Comment:
I think `SyncReportItem` is specific to google group sync. The `value` here probably needs to be different and constructed at the end of `case Some(message: PubSubMessage)`.